### DISCUSSION TITLE
Where can I find the implementation of OnPointerEnter event of TextBox?

### DISCUSSION THREAD

**Discussion by hopeko** [AvaloniaUI/Avalonia#5405] [Q&A] [answered]
I want to override OnPointerEnter event of TextBox, so I want to see the source code of it. But I can't find this method from TextBox.cs. I find out OnPointerEnter is a method defined in InputElement(interface) inherited by Control inherited by TextBox, but there's no such a method in Control.cs or TextBox.cs. I'm so confused.

**Comment by maxkatz6** [comment, marked as answer]
This method is called from PointerEnter handler
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Input/InputElement.cs#L205

Why do you need Control/TextBox overrides for this event?
You don't need to reimplement its logic in your method, it can be as simple as:
```csharp
protected virtual void OnPointerEnter(PointerEventArgs e)
{
    base.OnPointerEnter(e);

    // Your code.
}
```

**Reply by hopeko** [reply]
I just want to see the logit, then I can override it to change TextBox style properly. But later, I find out I can do that via Styles.
